### PR TITLE
Fix `CMS_add1_signer()` diagnostics, `EVP shake_ctrl()`, and doc nits

### PR DESCRIPTION
**Pull request by DDvO**
This fixes some empty error queue cases on unsupported algorithms with in `CMS_add1_signer()`
as (in part) requested by https://github.com/openssl/openssl/issues/11915#issuecomment-1756489351.

On this occasion, also 

* fix `EVP shake_ctrl()` by adding a missing NULL `evp_ctx` check - like `sha1_int_ctrl()`, the function must not choke on  a NULL argument. This is needed in particular for EdDSA support for CMS/PKCS7, as provided in #22391.
* in `CMS_add1_signer.pod`, add missing info on `CMS_SignerInfo_sign()` return values
* in `pkcs7.h.in`, point out confusing names of `digest_enc_alg` and `enc_digest` fields of `PKCS7_SIGNER_INFO`

### CODE REVIEW

**Comment by DDvO** [comment]
CI failure with [enable_brotli_dynamic](https://github.com/openssl/openssl/actions/runs/6520329486/job/17707775206?pr=22390#logs) must be unrelated.

File: doc/man3/CMS_add1_signer.pod [hunk 1]
```diff
@@ -31,7 +31,7 @@ Unless the B<CMS_REUSE_DIGEST> flag is set the returned CMS_ContentInfo
 structure is not complete and must be finalized either by streaming (if
 applicable) or a call to CMS_final().
 
-The CMS_SignerInfo_sign() function will explicitly sign a CMS_SignerInfo
+The CMS_SignerInfo_sign() function explicitly signs a CMS_SignerInfo
 structure, its main use is when B<CMS_REUSE_DIGEST> and B<CMS_PARTIAL> flags
```
Comment (vdukhovni) [inline doc/man3/CMS_add1_signer.pod:35]:
```suggestion
structure, its main use is when the B<CMS_REUSE_DIGEST> and B<CMS_PARTIAL> flags
```

File: doc/man3/CMS_add1_signer.pod [hunk 2]
```diff
@@ -90,6 +90,9 @@ before it is finalized.
 CMS_add1_signer() returns an internal pointer to the CMS_SignerInfo
 structure just added or NULL if an error occurs.
 
+CMS_SignerInfo_sign() returns 1 on success, 0 on failure.
+
+
```
Comment (vdukhovni) [inline doc/man3/CMS_add1_signer.pod:95]:
```suggestion

```

**Review by vdukhovni — APPROVED** [review]
Approval + cosmetic suggestions.

**Review by vdukhovni — APPROVED** [review]

**Comment by DDvO** [comment]
> Approval + cosmetic suggestions.

Thanks for your swift approval.
Nits in `CMS_add1_signer.pod` fixed as suggested.

**Review by t8m — APPROVED** [review]

**Comment by DDvO** [comment]
Merged to master, 3.0, and 3.1 - thanks @vdukhovni and @t8m for the approvals.

**Pull request closed without merging**